### ISSUE TITLE
Transform operations should target masks bounds instead of masked layer

### ISSUE THREAD

**Issue by chadrolfs** [adobe-photoshop/spaces-design#3172]
If you have 3 layers that have masks and you align or distribute or flip, the operation is performed on the layer bounds instead of mask.  Operations should be performed on the mask bounds but also keep the masked layer locked to the mask during the transformations.

This is an enhancement on standard PS, core work will probably be needed.

related to #3106

**Comment by pineapplespatula** [comment]
This would require DS to have multiple bounds on masked layers.

**Comment by chadrolfs** [comment]
Yeah, I don't think this will be an easy engineering feat, but what do I know.  From a user perspective, I'd want this functionality though.
